crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -290,6 +296,43 @@ impl TimelineSlice {
     }
 }
 
+#[allow(clippy::large_enum_variant)]
+/// Wrapper around both MemberEvent-Types
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum MemberEvent {
+    Stripped(StrippedRoomMemberEvent),
+    Full(OriginalSyncRoomMemberEvent),
```

Comment:
Why use a different naming than Ruma here? "Full" is used in a few other places to mean "including the room_id unlike events from sync".

Also, what about redacted ones? Could deal with that in a separate PR, but it's going to significantly alter what you added.

Reply:
> Why use a different naming than Ruma here? "Full" is used in a few other places to mean "including the room_id unlike events from sync".

You mean going for `Original` instead of `Full`? I don't really care. And I don't think it really matters (as this is mostly internal anyways).

> Also, what about redacted ones? Could deal with that in a separate PR, but it's going to significantly alter what you added.

Because this tries to fix the name calculation only - the PR is already a lot more complex than I expected this fix to be needed for. I don't really want to add more complexity by adding yet another aspect.

Sure, the other PR will alter this, that's fine. But after this PR we have tests in place that ensure whatever is being done, this problem stays fixed.